doc/developer/platform/formalism.md

```diff
@@ -747,12 +742,12 @@ The dataflow remains active until a corresponding `DROP INDEX` command is receiv
 The Adapter layer provides interactive transactions through the `BEGIN`, `COMMIT`, and `ROLLBACK` commands.
 There are various restrictions on these transactions, primarily because the Storage and Compute layers require you to durably write before you can read from them.
 This makes read-after-write transactions challenging to perform efficiently, and they are not currently supported.
-Write-after-read transactions are implemented by performing all reads at the current timestamp and then writing at a strictly later timestamp.
-Other read-only transactions may co-occur with a write-after-read transaction (logically preceding the transaction, as its writes are only visible at its write timestamp).
-Other write-only transactions may precede or follow a write-after-read transaction, but may not occur *between* the read and write times of that transaction.
-Multiple write-after-read transactions may not co-occur (their [read, write) half-open timestamp intervals must be disjoint).
+Read-then-write transactions are implemented by performing all reads at some timestamp greater than or equal to all previous timestamps and then writing at a strictly later timestamp.
```

Comment:
I think this text is still a bit confusing. It would be best if we can name the time at which the transaction appears to occur (the write timestamp, I believe), as well as additional constraints on the transaction. These are I believe that no updates to query inputs can occur at a time greater than the read timestamp and not greater than the write timestamp. This is currently only enforceable for table inputs (and views over tables, I think, but not ones that involve temporal filters?).

My sense at the moment is that we should maybe build up the transactions that are unconditionally supported (e.g. read only, write only), and follow with text about other forms of transaction that are supported and our best understanding of their rules.

Reply:
Let me know what you think of the update I made. I removed the two lines about co-occuring because in my opinion they are implementation details and are covered by the statement about not changing query inputs.